src/h3lib/lib/h3Index.c

```diff
@@ -214,6 +214,31 @@ void H3_EXPORT(h3ToChildren)(H3Index h, int childRes, H3Index* children) {
     }
 }
 
+/**
+ * h3ToCenterChild produces the center child index for a given H3 index at
+ * the specified resolution
+ *
+ * @param h H3Index to find center child of
+ * @param childRes The resolution to switch to
+ *
+ * @return H3Index of the center child, or 0 if you actually asked for a parent
+ */
+H3Index H3_EXPORT(h3ToCenterChild)(H3Index parentH, int childRes) {
+    int parentRes = H3_GET_RESOLUTION(parentH);
+    if (childRes < parentRes) {
+        return H3_INVALID_INDEX;
+    } else if (childRes == parentRes) {
+        return parentH;
+    } else if (childRes < 0 || childRes > MAX_H3_RES) {
```

Comment:
Nit 1: `childRes < 0` should not be possible if we've checked `childRes < parentRes`.
Nit 2: I'd combine this branch with the `if (childRes < parentRes)` branch above to consolidate validation.

More substantively - we need a similar check in `h3ToChildren` and `uncompact`, but it doesn't look like we check `MAX_H3_RES` there. Could we add a static function like `static bool isValidChildRes(int parentRes, int childRes)` and use in all three functions?